logic.ipynb

```diff
@@ -1438,55 +1438,520 @@
        "</html>\n"
       ],
       "text/plain": [
-       "<IPython.core.display.HTML object>"
+       "<IPython.core.display.HTML object>"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "psource(pl_resolution)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 38,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "(True, False)"
+      ]
+     },
+     "execution_count": 38,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "pl_resolution(wumpus_kb, ~P11), pl_resolution(wumpus_kb, P11)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 39,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "(False, False)"
+      ]
+     },
+     "execution_count": 39,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "pl_resolution(wumpus_kb, ~P22), pl_resolution(wumpus_kb, P22)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Forward and backward chaining\n",
+    "Introducing this section, we said we will look at two algorithms to check if a sentence is entailed by the `KB`, \n",
```

Comment:
I find the 'Introducing this section' a bit awkward. I suggest removing it. Something like 'Previously, we said ...' works best in my opinion.